core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -796,11 +802,12 @@ class DAGScheduler(
 
     // First figure out the indexes of partition ids to compute.
     val partitionsToCompute: Seq[Int] = {
-      if (stage.isShuffleMap) {
-        (0 until stage.numPartitions).filter(id => stage.outputLocs(id) == Nil)
-      } else {
-        val job = stage.resultOfJob.get
-        (0 until job.numPartitions).filter(id => !job.finished(id))
+      stage match {
+        case stage: ShuffleMapStage =>
+          (0 until stage.numPartitions).filter(id => stage.outputLocs(id) == Nil)
```

Comment:
Andrew - I think this one should be `.isEmpty` since it's looking for tasks that haven't completed. I've replaced other comparisons to Nil in the code.